torch/optim/functional.py

```diff
@@ -0,0 +1,46 @@
+r"""Functional interface"""
+from torch import Tensor
+from typing import List
+
+# TODO: use foreach API in optim.functional to do all the computation
```

Comment:
note to self: we'll need to make a conscious effort to avoid having too many versions of the same optimizers, with the risk of having them out of sync.

cc  #43507

Reply:
offline meeting with @wanchaol and @izdeby: #43507 will be merged first and then updated to functional form later.